hcipy/optics/segmented_mirror.py

```diff
@@ -0,0 +1,237 @@
+import os
+import numpy as np
+import matplotlib.pyplot as plt
+
+from .optical_element import OpticalElement
+from ..aperture import circular_aperture, hexagonal_aperture, make_segmented_aperture
+from ..field import Field, make_hexagonal_grid, make_pupil_grid, evaluate_supersampled
+from ..io import write_fits
+from ..plotting import imshow_field
+
+
+pupil_size = 1024
+PUP_DIAMETER = 15.   # m
```

Comment:
These need to go and either be an input parameter or inside the function/class.

Reply:
Agreed.

Reply:
`PUP_DIAMETER` is currently used in a couple of instances in `SegmentedMirror()`, specifically in the method `_setup_grids()`. It scales the segment gaps and creates a pupil grid for the aperture with the *numbered* segments. I'm unsure of how to get rid of this global variable as long as we need that scaling. See also my comment about the in-situ evaluation of the aperture.

Reply:
This has been resolved with the refactoring of the class for inputs of indexed apertures.